src/datasources/results/ResultsDataSource.ts

```diff
@@ -0,0 +1,131 @@
+import { DataFrameDTO, DataQueryRequest, DataSourceInstanceSettings, FieldType, TestDataSourceResponse } from '@grafana/data';
+import { BackendSrv, TemplateSrv, getBackendSrv, getTemplateSrv } from '@grafana/runtime';
+import { DataSourceBase } from 'core/DataSourceBase';
+import { OutputType, QueryResultsResponse, ResultsProperties, ResultsPropertiesOptions, ResultsQuery, ResultsResponseProperties } from './types';
+
+export class ResultsDataSource extends DataSourceBase<ResultsQuery> {
+  constructor(
+    readonly instanceSettings: DataSourceInstanceSettings,
+    readonly backendSrv: BackendSrv = getBackendSrv(),
+    readonly templateSrv: TemplateSrv = getTemplateSrv()
+  ) {
+    super(instanceSettings, backendSrv, templateSrv);
+  }
+
+  baseUrl = this.instanceSettings.url + '/nitestmonitor';
+  queryResultsUrl = this.baseUrl + '/v2/query-results';
+  defaultQuery = {
+    properties: [
+      ResultsPropertiesOptions.PROGRAM_NAME,
+      ResultsPropertiesOptions.PART_NUMBER,
+      ResultsPropertiesOptions.SERIAL_NUMBER,
+      ResultsPropertiesOptions.STATUS,
+      ResultsPropertiesOptions.HOST_NAME,
+      ResultsPropertiesOptions.STARTED_AT,
+      ResultsPropertiesOptions.UPDATED_AT,
+      ResultsPropertiesOptions.WORKSPACE
+    ] as ResultsProperties[],
+    outputType: OutputType.Data,
+    recordCount: 1000,
+  };
+
+  private timeRange: { [key: string]: string } = {
+    Started: 'startedAt',
+    Updated: 'updatedAt',
+  }
+  private fromDateString = '${__from:date}';
+  private toDateString = '${__to:date}';
+
+  async queryResults(
+    filter?: string,
+    orderBy?: string,
+    projection?: ResultsProperties[],
+    take?: number,
+    descending?: boolean,
+    returnCount = false
+  ): Promise<QueryResultsResponse> {
+    try {
+      return await this.post<QueryResultsResponse>(`${this.queryResultsUrl}`, {
+        filter,
+        orderBy,
+        descending,
+        projection,
+        take,
+        returnCount,
+      });
+    } catch (error) {
+      throw new Error(`An error occurred while querying results: ${error}`);
+    }
+  }
+
+  private getVariableReplacedFilter(query: ResultsQuery, options: DataQueryRequest): string | undefined {
+    let queryByFilter = undefined;
+
+    if (query.useTimeRange === false) {
```

Comment:
Simplify this logic  something like
```suggestion
   if (query.useTimeRange === true && query.useTimeRangeFor !== undefined) {
    const timeFilter = `(${this.timeRange[query.useTimeRangeFor]} > \"${this.fromDateString}\" && ${this.timeRange[query.useTimeRangeFor]} < \"${this.toDateString}\")`;
   return  this.templateSrv.replace(timeFilter, options.scopedVars) ;
   }
   return undefined;
```

Reply:
Updated